src/app/(pages)/(TwoColumn)/article/[id]/page.jsx

```diff
@@ -0,0 +1,39 @@
+import DetailBody from "@/components/section/twoColumn/DetailBody";
+import { getDetails } from "@/components/common/fetchData";
+import {
+  Banner,
+  Breadcrumb,
+  PageTitle,
+  TagArea,
+  TagKeyword,
+} from "@/components/common";
+import Feature from "@/components/section/feature/Feature";
+
+export default async function Food({ params }) {
+  const item = await getDetails(params.id);
+
+  const content = {
```

Comment:
@yabe-diverta san

According to https://nextjs.org/docs/app/building-your-application/caching#request-memoization

Do you think we can solve the problem of memoizing by just removing the `{ cache: "no-store" }` line from every fetch command inside [fetchData.js](https://github.com/diverta/front_next_media/pull/5/files#diff-5b3a7c3287ce696355fe5bbb747e3259da918166422a94978e2e032d8f655130R21) and not using the react useMemo hook since memoizing the fetch calls is a default behaviour in Next 13 

What do you think ? Am I missing something ? 🙇

Reply:
No it's not fetch stuff.
I meant it was not a good way to define object in a component,
because it would consume memory on every re-rendering.

like:
```js
  const content = {
    text: '記事詳細',
    text_en: 'Article details',
  };
```
should be:
```js
  const content = useMemo({
    text: '記事詳細',
    text_en: 'Article details',
  }, []);
```
useMemo() or useRef() to wrapup data.

The `content` data seems too light weight to consider memory usage in a sense,
although wrapping up is standard way for many React ppl.
So it's easier to wrap such any data in every components rather than thinking deeply either is better in every components.

Reply:
@yabe-diverta san

Same as https://github.com/diverta/front_next_media/pull/5#discussion_r1456779076
What I tried : `src/app/(pages)/(TwoColumn)/article/[id]/page.jsx` replaced by code in [page.txt](https://github.com/diverta/front_next_media/files/13971821/page.txt)

Using useMemo would require changing to client component from server. 
Entering infinite loop of console statements due to multiple re-renders. 

Would prefer leaving it as a server component because I believe the content is not a big overhead and is not rendered multiple times during navigation

Once I load any of the pages let's say article/1 it is stored in the cache by default, if I go to article/2 and then come back to article/1 it will not fetch the content value or the api again but rather just load it from the cache. So using useMemo and converting it to client should not be necessary according to me. 

Please check and let me know what you think. 🙇

Reply:
> Using useMemo would require changing to client component from server.

Yes it's not good to change to client component as well.
please ignore my comment 🙇